galley/pkg/config/analysis/analyzers/virtualservice/util.go

```diff
@@ -17,34 +17,76 @@ import (
 	"istio.io/api/networking/v1alpha3"
 )
 
-func getRouteDestinations(vs *v1alpha3.VirtualService) []*v1alpha3.Destination {
-	destinations := make([]*v1alpha3.Destination, 0)
+// Destination is created because the index of the destination along with the destination object are needed
+// for calculating the destination path in the YAML resource
+type Destination struct {
+	routeRule        string
+	serviceIndex     int
+	destinationIndex int
+	destination      *v1alpha3.Destination
+}
+
+func (d *Destination) GetRouteRule() string {
+	return d.routeRule
+}
+
+func (d *Destination) GetServiceIndex() int {
+	return d.serviceIndex
+}
+
+func (d *Destination) GetDestinationIndex() int {
+	return d.destinationIndex
+}
+
+func (d *Destination) GetDestination() *v1alpha3.Destination {
+	return d.destination
+}
```

Comment:
Unless there's a strong reason to not export the properties, it would be simpler to just export them and get rid of the getter functions. If there's a strong reason, then please remove "Get" from the function names, see https://golang.org/doc/effective_go.html#Getters